code/modules/atmospherics/machinery/pipes/pipe_spritesheet_helper.dm

```diff
@@ -0,0 +1,148 @@
+/client/proc/GeneratePipeSpritesheet()
+	set name = "Generate Pipe Spritesheet"
+	set category = "Debug"
+
+	var/datum/pipe_icon_generator/generator = new
+	generator.Start()
+	fcopy(generator.generated_icons, "icons/obj/pipes_n_cables/!pipes_bitmask.dmi")
+
+	generator.Start("-gas")
+	fcopy(generator.generated_icons, "icons/obj/pipes_n_cables/!pipe_gas_overlays.dmi")
+
+/datum/pipe_icon_generator
+	var/static/icon/template_pieces = icon('icons/obj/pipes_n_cables/pipe_template_pieces.dmi')
+	var/static/list/icon/damage_masks = list(
+		"[NORTH]"=icon('icons/obj/pipes_n_cables/pipe_template_pieces.dmi', "damage_mask", NORTH),
+		"[EAST]"=icon('icons/obj/pipes_n_cables/pipe_template_pieces.dmi', "damage_mask", EAST),
+		"[SOUTH]"=icon('icons/obj/pipes_n_cables/pipe_template_pieces.dmi', "damage_mask", SOUTH),
+		"[WEST]"=icon('icons/obj/pipes_n_cables/pipe_template_pieces.dmi', "damage_mask", WEST),
+	)
+
+	var/icon/generated_icons
+
+/datum/pipe_icon_generator/proc/Start(icon_state_suffix="")
+	var/list/outputs = list()
+	for(var/layer in 1 to 5)
+		// Since dirs are bitflags, if we want to iterate over every possible direction
+		// combination we just need to iterate over every number that can be contained in 4 bits.
+		//
+		// I wrote this all the hard way originally >.>
+		for(var/combined_dirs in 1 to 15)
+			switch(combined_dirs)
+				if(NORTH, EAST, SOUTH, WEST)
+					continue
+
+			outputs += GeneratePipeDir(icon_state_suffix, layer, combined_dirs)
+
+	generated_icons = icon('icons/testing/greyscale_error.dmi')
+	for(var/icon/generated_icon as anything in outputs)
+		var/pending_icon_state = outputs[generated_icon]
+		generated_icons.Insert(generated_icon, pending_icon_state)
+
+/datum/pipe_icon_generator/proc/GeneratePipeDir(icon_state_suffix, layer, combined_dirs)
```

Comment:
Is there a reason we're switching to pascal case for all these procs?

Reply:
That's how I've written procs when working on new code for a while now, and we don't have naming conventions. I only match nearby style if there is a style to match.

Reply:
huh I guess that line in the style guide actually is only about typepaths, never mind then

Reply:
oranges tried to make it cover procs too but burned out on it. so not a bad instinct